src/makielayout/layoutables/colorbar.jl

```diff
@@ -78,7 +78,8 @@ function layoutable(::Type{<:Colorbar}, fig_or_scene; bbox = nothing, kwargs...)
         if all(!isnothing, (limits, colorrange))
             error("Both colorrange + limits are set, please only set one, they're aliases. colorrange: $(colorrange), limits: $(limits)")
         end
-        return something(limits, colorrange, (0, 1))
+        lo, hi = lims = something(limits, colorrange, (0, 1))
+        return lo == hi ? (zero(lo), one(hi)) : lims
```

Comment:
shouldn't this be something like `(hi - 0.5, hi + 0.5)` ?

Reply:
Oh, yes I forgot about centering, sorry, will fix. I also tried to fix this after in the pipeline (in `src/makielayout/lineaxis.jl`), but without success.

Reply:
Oddly, when applying `(oftype(lo, lo - .5), oftype(hi, hi + .5))` instead, I've got this

![z_rng](https://user-images.githubusercontent.com/13423344/157909379-d0d88b3f-36b1-4ae3-9558-bde53cd336c5.png)

, which I'm afraid is obviously wrong.

Reply:
the heatmap also needs the colorrange

Reply:
what you did is build a colorbar that takes its colormap from the heatmap but sets different limits, without synching those back to the heatmap.

Reply:
Thanks for pointing that out. Here is the new output:
![hm-mid](https://user-images.githubusercontent.com/13423344/158030173-7ff489c3-adec-4d6a-a764-66b5cc69d1f2.png)